server/src/main/java/org/elasticsearch/index/SearchSlowLog.java

```diff
@@ -118,7 +127,8 @@ public final class SearchSlowLog implements SearchOperationListener {
 
     private static final ToXContent.Params FORMAT_PARAMS = new ToXContent.MapParams(Collections.singletonMap("pretty", "false"));
 
-    public SearchSlowLog(IndexSettings indexSettings) {
+    public SearchSlowLog(IndexSettings indexSettings, SlowLogFieldProvider slowLogFieldProvider) {
+        slowLogFieldProvider.init(indexSettings);
```

Comment:
Needs an init call here to enable the security plugin to listen for updates to the settings and set the initial values.